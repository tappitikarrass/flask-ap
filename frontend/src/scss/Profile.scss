@use "base16" as *;


#profile {
    display: flex;
    flex-direction: column;
    justify-content: center;
}

#bt-row.ui-row {
    margin-top: 1em;
    & > button,
    & > .bt-blu {
        width: 100%;
    }
}

th {
    text-align: right;
    user-select: none;
}

// phone
@media screen and (max-width: 500px) {
    th {
        width: 25vw;
    }
    #bt-row {
        gap: 1em;
    }
}

// tablet
@media screen and (min-width: 500px) and (max-width: 810px) {
    th {
        width: 15vw;
    }
}

// desktop
@media screen and (min-width: 810px) {
    th {
        width: 10vw;
    }
}
