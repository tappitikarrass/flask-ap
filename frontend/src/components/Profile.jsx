import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useCookies } from 'react-cookie';
import '../scss/Profile.scss';

function Profile() {
  const [userData, setUserData] = useState([]);
  const [cookies, removeCookie] = useCookies(['token', 'username', 'user_id']);
  const navigate = useNavigate();

  /* istanbul ignore next */
  async function logOut() {
    if (cookies.token == null) {
      return;
    }

    const response = await fetch(
      'http://localhost/backend/logout',
      {
        method: 'DELETE',
        headers: {
          Authorization: `Bearer ${cookies.token}`,
        },
      },
    );
    /* istanbul ignore next */
    if (response.status === 200) {
      removeCookie('token');
      removeCookie('username');
      removeCookie('user_id');
      navigate('/login');
    }
  }

  /* istanbul ignore next */
  async function deleteUser() {
    if (cookies.token == null) {
      return;
    }

    const userName = cookies.username;
    const response = await fetch(
      `http:///localhost/backend/user/${userName}`,
      {
        method: 'DELETE',
        headers: {
          Authorization: `Bearer ${cookies.token}`,
        },
      },
    );

    if (response.status === 200) {
      removeCookie('token');
      removeCookie('username');
      navigate('/login');
    }
  }

  useEffect(() => {
    async function effect() {
      const response = await fetch(
        `http://localhost/backend/user/${cookies.username}`,
        {
          method: 'GET',
          headers: {
            Authorization: `Bearer ${cookies.token}`,
          },
        },
      );
      /* istanbul ignore next */
      setUserData(await response.json());
    }
    /* istanbul ignore next */
    if (cookies.token === 'undefined') {
      navigate('/login');
    } else {
      effect();
    }
  });

  return (
    <div id="profile" className="content">
      <h2>Profile</h2>
      <table>
        <tbody>
          <tr>
            <th>Username</th>
            <td id="username">
              {userData.username}
            </td>
          </tr>
          <tr>
            <th>First name</th>
            <td id="firstname">
              {userData.firstname}
            </td>
          </tr>
          <tr>
            <th>Last name</th>
            <td id="lastname">
              {userData.lastname}
            </td>
          </tr>
          <tr>
            <th>Phone</th>
            <td id="phone">
              {userData.phone}
            </td>
          </tr>
          <tr>
            <th>Email</th>
            <td id="email">
              {userData.email}
            </td>
          </tr>
        </tbody>
      </table>
      <div id="bt-row" className="ui-row">
        <button
          type="submit"
          className="bt-blue"
          onClick={() => { 
            /* istanbul ignore next */
            navigate('/profile/edit');
          }}
        >
          Edit
        </button>
        <button
          type="submit"
          className="bt-red"
          onClick={deleteUser}
        >
          Delete
        </button>
        <button
          type="submit"
          className="bt-green-end"
          onClick={logOut}
        >
          Logout
        </button>
      </div>
    </div>
  );
}

export default Profile;
