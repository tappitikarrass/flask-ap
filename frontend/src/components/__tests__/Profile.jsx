import React from "react";
import { HashRouter } from "react-router-dom";
import { create, act } from "react-test-renderer";
import Profile from '../Profile';

describe('Profile', () => {
  it('',() => {
    let component;
    act(() => {
      component = create(
        <HashRouter>
          <Profile />
        </HashRouter>
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
