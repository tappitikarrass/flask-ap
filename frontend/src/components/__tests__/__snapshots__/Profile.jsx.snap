// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile  1`] = `
<div
  className="content"
  id="profile"
>
  <h2>
    Profile
  </h2>
  <table>
    <tbody>
      <tr>
        <th>
          Username
        </th>
        <td
          id="username"
        />
      </tr>
      <tr>
        <th>
          First name
        </th>
        <td
          id="firstname"
        />
      </tr>
      <tr>
        <th>
          Last name
        </th>
        <td
          id="lastname"
        />
      </tr>
      <tr>
        <th>
          Phone
        </th>
        <td
          id="phone"
        />
      </tr>
      <tr>
        <th>
          Email
        </th>
        <td
          id="email"
        />
      </tr>
    </tbody>
  </table>
  <div
    className="ui-row"
    id="bt-row"
  >
    <button
      className="bt-blue"
      onClick={[Function]}
      type="submit"
    >
      Edit
    </button>
    <button
      className="bt-red"
      onClick={[Function]}
      type="submit"
    >
      Delete
    </button>
    <button
      className="bt-green-end"
      onClick={[Function]}
      type="submit"
    >
      Logout
    </button>
  </div>
</div>
`;
